### ISSUE TITLE
I don't believe this gem correctly migrates passwords from bcrypt

### ISSUE THREAD

**Issue by swrobel** [venables/has_secure_password_argon2#14]
It's my understanding from reading the code that when it correctly authenticates a bcrypt password, it migrates it to argon2 by setting [`self.password = hash_argon2_password(unencrypted_password)`](https://github.com/venables/has_secure_password_argon2/blob/master/lib/has_secure_password_argon2.rb#L42) which in turn calls the overridden [`password=`](https://github.com/venables/has_secure_password_argon2/blob/master/lib/has_secure_password_argon2.rb#L64), which again calls `hash_argon2_password()`, thus hashing double-hashing the password and preventing it from authenticating correctly on the next login.

You can verify this by changing this test case to add a second assertion making sure the password authenticates correctly after changing it to argon2:
```ruby
test '#authenticate updates the model for argon2 and returns the model upon valid bcrypt password' do
  password = 'testing'
  bcrypt_hash = BCrypt::Password.create(password)

  model = Model.new(password_digest: bcrypt_hash)
  assert_changes -> { model.password_digest } do
    assert model.authenticate(password)
  end

  assert model.password_digest.start_with?('$argon2')
  assert model.authenticate(password) # NEW ASSERTION
  assert model.persisted?
end
```

This second assertion will fail. This should be fixed, or if this gem is unmaintained as it appears to be, it should be yanked from rubygems to avoid anyone else using it in the future.

**Comment by venables** [comment]
Thanks for catching this @swrobel.  Should be fixed in #15 .

For future reference, saying an open source project should yank from rubygems because of a bug in an upgrade path is not really appropriate, considering people are actively using the package as-is.  Please consider making a PR to move the project forward in those cases.

**Comment by swrobel** [comment]
Thanks for the quick fix. I realize yanking is a drastic move, but I was
only suggesting it in the case that you had no intention of maintaining
this gem, which seemed like a reasonable assumption given that the PR to
add compatibility with Rails 6 has been open for 2.5 years.
